src/Components/ReusedButtons.jsx/ArticleVoteButton.jsx

```diff
@@ -1,7 +1,73 @@
-export default function ArticleVoteButton() {
+import { useState, useEffect } from "react";
+import { patchVoteById } from "../../api";
+
+export default function ArticleVoteButton({ article_id, votes }) {
+  const [voteCount, setVoteCount] = useState(votes);
+  const [vote, setVote] = useState(0);
+  const [likeId, setLikeId] = useState("neutral");
+  const [dislikeId, setDislikeId] = useState("neutral");
+
+  const handleLikeClick = (event) => {
+    if (likeId === "neutral") {
+      setLikeId("liked");
+
+      console.log(likeId, dislikeId);
+
+      if (dislikeId === "disliked") {
+        setVote((vote) => vote + 2);
+      } else {
+        setVote((vote) => vote + 1);
+      }
+      setDislikeId("neutral");
+    } else if (likeId === "liked") {
+      setLikeId("neutral");
+      setVote((vote) => vote - 1);
+    }
+  };
+
+  const handleDislikeClick = (event) => {
+    if (dislikeId === "neutral") {
+      setDislikeId("disliked");
+      if (likeId === "liked") {
+        setVote((vote) => vote - 2);
+      } else {
+        setVote((vote) => vote - 1);
+      }
+      setLikeId("neutral");
+    } else if (dislikeId === "disliked") {
+      setDislikeId("neutral");
+      setVote((vote) => vote + 1);
+    }
+  };
+
+  useEffect(() => {
+    let body = { inc_vote: vote };
+    patchVoteById(article_id, body).then((res) => {
+      console.log(res);
+      setVoteCount(res);
+      setVote(0);
+    });
+  }, [vote]);
+
   return (
     <div className="article-vote-button">
-      <h5>article vote button</h5>
+      <span>
+        <input
+          type="button"
+          className="like-btn"
+          id={likeId}
+          onClick={handleLikeClick}
+          value="Y"
```

Comment:
I think the naming of buttons - "Y" and "N" is confusing, I would re-think what you would want to name this so it is semantic. I would also think about the styling as at the moment it is overlapping for me.